instantsearch/src/main/java/com/algolia/instantsearch/model/SearchResults.java

```diff
@@ -160,6 +164,52 @@ public SearchResults(@NonNull JSONObject content) {
         }
     }
 
+    public SearchResults(@NonNull JSONObject content, Map<String, List<String>> hierarchicalFacetDeclarations) {
+        this(content);
+        this.hierarchicalFacetDeclarations = hierarchicalFacetDeclarations;
+    }
+
+    @NonNull public List<HierarchicalFacetValue> getHierarchicalFacet(@NonNull String name) {
+        return getHierarchicalFacet(name, " > ");
+    }
+
+    @NonNull public List<HierarchicalFacetValue> getHierarchicalFacet(@NonNull String name, @NonNull final String separator) {
+        List<String> attributes = hierarchicalFacetDeclarations.get(name);
+        if (attributes == null || attributes.isEmpty()) {
+            return new ArrayList<>();
```

Comment:
I'd rather have `getHierarchicalFacet` be `@Nullable` , to be consistent with 
```java
@Nullable List<String> getFacetRefinements(@NonNull String attribute)
```` 
and 
```java
@Nullable NumericRefinement getNumericRefinement(@NonNull String attribute, int operator)
```